src/CodeGenerators/SettingsGen/BaseGenerator.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.Omex.CodeGenerators.SettingsGen.FileGeneration;
+using Microsoft.Omex.CodeGenerators.SettingsGen.Parser;
+using Microsoft.Omex.CodeGenerators.SettingsGen.Validation;
+
+namespace Microsoft.Omex.CodeGenerators.SettingsGen
+{
+	/// <summary>
+	/// The base generator which can be inherited in order to implement your own version
+	/// </summary>
+	/// <typeparam name="TSettingModel">Settings model type</typeparam>
+	public abstract class BaseGenerator<TSettingModel> : ISourceGenerator
```

Comment:
Since we are not generating source code should not it be an analyzer instead of a source generator?
I understand it's essentially the same, but it could help better express intention.

Reply:
I think I will leave it as a generator as it makes it clearer that its generating files and also current;y the code is using source generator interfaces and structs, so would require a big rewrite to get it working again, so will leave it as a generator